runtime/vm/BufferWriter.hpp

```diff
@@ -203,19 +238,21 @@ class VM_BufferWriter {
 	void
 	writeLEB128(U_64 val)
 	{
-		U_64 newVal = val;
-		if (!_isLE) {
-			newVal = byteSwap(val);
-		}
-		do {
-			U_8 byte = newVal & 0x7F;
-			newVal >>= 7;
-
-			if (newVal > 0) {
-				byte |= 0x80;
+		if (checkBounds(9)) {
+			U_64 newVal = val;
+			if (!_isLE) {
+				newVal = byteSwap(val);
 			}
```

Comment:
I don't think calling `byteSwap()` is appropriate for any platform here.

Reply:
Ill address this